assets/js/components/SideSheet.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * SideSheet component
+ *
+ * This component is named after Material Side Sheets
+ * (https://m3.material.io/components/side-sheets/overview), which is
+ * planned but not yet implemented for Web. Once available, we could look
+ * into moving this to `googlesitekit-components` and replacing it with
+ * the Material implementation.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Portal from './Portal';
+
+export default function SideSheet( { className, children, isOpen } ) {
+	// Disable scrolling on document body when panel is open.
+	useEffect( () => {
+		if ( isOpen ) {
+			document.body.style.overflow = 'hidden';
+		} else {
+			document.body.style.removeProperty( 'overflow' );
+		}
+	}, [ isOpen ] );
+
+	return (
+		<Portal>
+			<section
+				className={ classnames( 'googlesitekit-side-sheet', className, {
+					'googlesitekit-side-sheet--open': isOpen,
+				} ) }
+			>
```

Comment:
How about we use [`aria-modal` with a role of `dialog`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal)? Even though this isn't technically a modal, it does act like one, at least to screen readers. What do you think?

Reply:
Yeah, I'm not sure which is most appropriate (or both?) but it seems like some indication is needed.

Reply:
Both is the way to go for this one, especially as it'll be grabbing focus when opened, right?

When hidden we should do `aria-hidden`, when visible the `aria-modal="true"` + `role="dialog"` works. It's functionally a modal I'd say, so I think it's fine to label it as one.

👍🏻

Reply:
Applied accordingly, thank you!

Just a note that following the dashboard sharing settings dialog, I've left `aria-modal="true"` + `role="dialog" present both when it is hidden and visible, but adjusted `aria-hidden` to be `true` only when visible. Let me know if I should do it otherwise.

Thank you!